src/lxc/console.c

```diff
@@ -733,8 +736,8 @@ int lxc_console(struct lxc_container *c, int ttynum,
 	close(masterfd);
 	close(ttyfd);
 err1:
-	tcsetattr(stdinfd, TCSAFLUSH, &oldtios);
+	if (istty)
+		tcsetattr(stdinfd, TCSAFLUSH, &oldtios);
```

Comment:
Can you also issue an `INFO()` or `WARN()` including the output from `strerror(errno)` right here in case `tcsetattr()` failed. Might be useful in debugging.